### PR TITLE
Controller mapping

### PR DESCRIPTION
**Pull request by yawhide**

### CODE REVIEW

**Comment by yawhide** [comment]
@dvdfu @akwpark

File: controller.lua [hunk 1]
```diff
@@ -3,6 +3,9 @@ Controller = class('Controller')
 function Controller:initialize(i)
 	self.controller = i
 	self.joystick = love.joystick.getJoysticks()[i]
+	self.inputtype, self.rbumper = self.joystick:getGamepadMapping('rightshoulder')
```
Comment (dvdfu) [inline controller.lua]: `camelCase`

File: controller.lua [hunk 2]
```diff
@@ -3,6 +3,9 @@ Controller = class('Controller')
...
+	self.vibration = self.joystick:isVibrationSupported()
```
Comment (dvdfu) [inline controller.lua]: `self.canVibrate`, maybe?

File: main.lua [hunk 3]
```diff
@@ -10,10 +10,10 @@ function love.load()
 	math.randomseed(os.time())
 	love.graphics.setDefaultFilter('nearest', 'nearest')
 	love.graphics.setBackgroundColor(40, 50, 60)
-
+	local joystickcount = love.joystick.getJoystickCount( )
```
Comment (dvdfu) [inline main.lua]: also camel

File: controller.lua (hunk 1 above)
Comment (yawhide) [inline controller.lua]: oh how times have changed

File: main.lua [hunk 4]
```diff
@@ -40,6 +40,11 @@ function love.update(dt)
 end
 
 function love.draw()
+	local joysticks = love.joystick.getJoysticks()
```
Comment (dvdfu) [inline main.lua:43]: `getJoysticks()` is needlessly queried every frame - we should just store this as a variable

File: main.lua [hunk 5]
```diff
@@ -40,6 +40,11 @@ function love.update(dt)
...
+	for i, joystick in ipairs(joysticks) do
+		love.graphics.print(joystick:getName(), 10, i * 20)
```
Comment (dvdfu) [inline main.lua:45]: oh, is this just for debugging? nvm

Comment (yawhide) [inline main.lua:45]: yes

File: controller.lua [hunk 6]
```diff
@@ -3,6 +3,15 @@ Controller = class('Controller')
 function Controller:initialize(i)
 	self.controller = i
 	self.joystick = love.joystick.getJoysticks()[i]
+	self.inputType, self.rBumper = self.joystick:getGamepadMapping('rightshoulder')
+	self.canVibrate = self.joystick:isVibrationSupported()
+	self.id = self.joystick:getID()
+
+	if self.canVibrate then
+		left, right = self.joystick:getVibration( )
```
Comment (dvdfu) [inline controller.lua]: they should be self

**Comment by akwpark** [comment]
Looks good

**Pull request merged**